cpp/src/fil/treelite_import.cu

```diff
@@ -69,32 +69,64 @@ int tree_root(const tl::Tree<T, L>& tree)
   return 0;  // Treelite format assumes that the root is 0
 }
 
-template <typename T, typename L>
-inline int max_depth(const tl::Tree<T, L>& tree)
+using empty = struct {
+};
+
+template <typename T,
+          typename L,
+          typename DescentAccumulator,
+          typename InnerFunc,
+          typename LeafFunc>
+inline void walk_tree(const tl::Tree<T, L>& tree,
+                      DescentAccumulator desc_acc,
```

Comment:
Also, in every use of this function, the initial value is either `0` or `empty{}`, i.e. the default value of its type. Consider dropping the parameter altogether, or providing it with the default value.